Modules/termios.c

```diff
@@ -315,6 +315,141 @@ termios_tcflow_impl(PyObject *module, int fd, int action)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+termios.tcgetwinsize
+
+    fd: fildes
+    /
+
+Get the tty winsize for file descriptor fd.
+
+Returns a list [ws_row, ws_col].
+[clinic start generated code]*/
+
+static PyObject *
+termios_tcgetwinsize_impl(PyObject *module, int fd)
+/*[clinic end generated code: output=31825977d5325fb6 input=c7ed8aa957d108c0]*/
+{
+#if defined(TIOCGWINSZ)
+    termiosmodulestate *state = PyModule_GetState(module);
+    struct winsize w;
+    if (ioctl(fd, TIOCGWINSZ, &w) == -1) {
+        return PyErr_SetFromErrno(state->TermiosError);
+    }
+
+    PyObject *v;
+    if (!(v = PyList_New(2))) {
+        return NULL;
+    }
+
+    PyList_SetItem(v, 0, PyLong_FromLong((long)w.ws_row));
+    PyList_SetItem(v, 1, PyLong_FromLong((long)w.ws_col));
+    if (PyErr_Occurred()) {
+        Py_DECREF(v);
+        return NULL;
+    }
+    return v;
+#elif defined(TIOCGSIZE)
+    termiosmodulestate *state = PyModule_GetState(module);
+    struct ttysize s;
+    if (ioctl(fd, TIOCGSIZE, &s) == -1) {
+        return PyErr_SetFromErrno(state->TermiosError);
+    }
+
+    PyObject *v;
+    if (!(v = PyList_New(2))) {
+        return NULL;
+    }
+
+    PyList_SetItem(v, 0, PyLong_FromLong((long)s.ts_lines));
+    PyList_SetItem(v, 1, PyLong_FromLong((long)s.ts_cols));
+    if (PyErr_Occurred()) {
+        Py_DECREF(v);
+        return NULL;
+    }
+    return v;
+#else
+    PyErr_SetString(PyExc_NotImplementedError,
+                    "requires termios.TIOCGWINSZ and/or termios.TIOCGSIZE");
+    return NULL;
+#endif /* defined(TIOCGWINSZ) */
+}
+
+/*[clinic input]
+termios.tcsetwinsize
+
+    fd: fildes
+    winsize as winsz: object
+    /
+
+Set the tty winsize for file descriptor fd.
+
+The winsize to be set is taken from the winsize argument, which
+is a list like the one returned by tcgetwinsize().
```

Comment:
rather than referring to tcgetwinsize, also describe the format.  so someone reading this does not need to go look elsewhere to get the information.  "a two item tuple of (rows, cols) similar to what tcgetwinsize() returns" perhaps.